openmc/mgxs/mgxs.py

```diff
@@ -1884,13 +1886,19 @@ def tallies(self):
             energy = openmc.Filter('energy', group_edges)
             energyout = openmc.Filter('energyout', group_edges)
 
-            # Create a list of scores for each Tally to be created
-            if self.correction == 'P0':
-                scores = ['flux', 'scatter', 'scatter-P1']
-                filters = [[energy], [energy, energyout], [energyout]]
-            else:
-                scores = ['flux', 'scatter']
-                filters = [[energy], [energy, energyout]]
+            # Create lists of scores, filters for each Tally to be created
+            scores = ['flux']
+            filters = [[energy]]
+
+            # Create separate tallies for each moment
+            for moment in range(self.legendre_order+1):
+                scores.append('scatter-{}'.format(moment))
```

Comment:
Alright - any issues if I get around this issue by making my tally merging
code "smarter" and merging "scatter-X" scores into a single "scatter-PN"
score?
On May 11, 2016 2:52 PM, "Adam Nelson" <notifications@github.com> wrote:

> In openmc/mgxs/mgxs.py
> <https://github.com/mit-crpg/openmc/pull/644#discussion_r62904947>:
>
> > @@ -1884,13 +1886,19 @@ def tallies(self):
> >              energy = openmc.Filter('energy', group_edges)
> >              energyout = openmc.Filter('energyout', group_edges)
> >
> > -            # Create a list of scores for each Tally to be created
> > -            if self.correction == 'P0':
> > -                scores = ['flux', 'scatter', 'scatter-P1']
> > -                filters = [[energy], [energy, energyout], [energyout]]
> > -            else:
> > -                scores = ['flux', 'scatter']
> > -                filters = [[energy], [energy, energyout]]
> > +            # Create lists of scores, filters for each Tally to be created
> > +            scores = ['flux']
> > +            filters = [[energy]]
> > +
> > +            # Create separate tallies for each moment
> > +            for moment in range(self.legendre_order+1):
> > +                scores.append('scatter-{}'.format(moment))
>
> It is true that "scatter-PN" is more efficient than "scatter-X". The
> cycling through the orders is an inner loop for "scatter-PN", while if you
> had "scatter-0 scatter-1 scatter-2 ..." then you have to cycle through the
> score loop each time and incur that overhead.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/openmc/pull/644/files/a6fd59a6227cdf2ae29c0e71778d457e335b0bcf#r62904947>
>